R/uniDimBayesianOutputPlots.R

```diff
@@ -0,0 +1,389 @@
+
+
+.BayesianPosteriorPlot <- function(jaspResults, model, options) {
+
+  if (!is.null(.getStateContainerB(jaspResults)[["plotContainer"]]$object))
+    return()
+
+  plotContainer <- createJaspContainer(gettext("Posterior Plots"))
+  plotContainer$dependOn(options = c("plotPosterior", "shadePlots", "probTable", "probTableValueLow",
+                                     "probTableValueHigh", "fixXRange", "dispPrior", "credibleIntervalValueScale",
+                                     "alphaScale", "omegaScale", "lambda2Scale", "lambda6Scale", "glbScale"))
+
+  derivedOptions <- model[["derivedOptions"]]
+  selected <- derivedOptions[["selectedEstimatorsPlots"]]
+  indices   <- which(selected)
+  nmsLabs   <- derivedOptions[["namesEstimators"]][["plots"]]
+  nmsObjs   <- derivedOptions[["namesEstimators"]][["tables"]]
+  nmsObjsNoGreek   <- derivedOptions[["namesEstimators"]][["plotsNoGreek"]]
+  idMatchedNames <- which(!is.na(charmatch(names(model), names(selected[indices]))))
+
+  if (options[["shadePlots"]] && options[["probTable"]])
+    shadePlots <- c(options[["probTableValueLow"]], options[["probTableValueHigh"]])
+  else
+    shadePlots <- NULL
+
+  if (options[["plotPosterior"]] && is.null(model[["empty"]])) {
+    n.item <- model[["k"]]
+    priors <- Bayesrel:::priors[[as.character(n.item)]]
+    # the prior names dont match the model names, thus rename the priors in their original order
+    names(priors) <- c("alphaScale", "lambda2Scale", "lambda6Scale", "glbScale", "omegaScale")
+
+    z <- 1
+    for (i in indices) {
+      if (is.null(plotContainer[[nmsObjsNoGreek[i]]])) {
+        prior <- priors[[grep(names(model)[idMatchedNames[z]], names(priors))]]
+        p <- .makeSinglePosteriorPlot(model[[idMatchedNames[z]]], nmsLabs[[i]],
+                                                         options[["fixXRange"]], shadePlots,
+                                                         options[["dispPrior"]], prior)
```

Comment:
don't you have these names already? 
```suggestion
        nm <- names(indices)[i]
        prior <- priors[[nm]]
        p <- .makeSinglePosteriorPlot(model[[nm, nmsLabs[[i]],
                                      options[["fixXRange"]], shadePlots,
                                      ptions[["dispPrior"]], prior)
```
edit: I noticed that `averageInterItemCor`, `meanScale`, and `sdScale` have different names in the model than in the derivedOptions. Why not ensure that they have the same names?

Reply:
Well I do have to match the names from the model with the selected coefficients, dont I? I guess if the names in the model are the same as the options, I could use something other than charmatch at least.

Reply:
the names of `indices` **are** the names of the selected coefficients! So `names(which(indices))` should give you those names. Except for `averageInterItemCor` and the others, but there I'd suggest unifying the names.

Reply:
okay. but how do I then check at what position in the `model` object the relevant results are, and need to be filled into the table? Just because I have the names, doesnt mean I know the position, and the position in the list is not always the same, thats why I did this whole matching thing

Reply:
I've left an example below